src/librustc_trans/trans/expr.rs

```diff
@@ -1767,7 +1768,37 @@ fn trans_eager_binop<'blk, 'tcx>(bcx: Block<'blk, 'tcx>,
       }
       ast::BiRem => {
         if is_float {
-            FRem(bcx, lhs, rhs, binop_debug_loc)
+            // LLVM currently always lowers the `frem` instructions appropriate
+            // library calls typically found in libm. Notably f64 gets wired up
+            // to `fmod` and f32 gets wired up to `fmodf`. Conveniently for us,
+            // 32-bit MSVC does not actually have a `fmodf` symbol, it's instead
+            // just an inline function in a header that goes up to a f64, uses
+            // `fmod`, and then comes back down to a f32.
+            //
+            // Although LLVM knows that `fmodf` doesn't exist on MSVC, it will
+            // still unconditionally lower frem instructions over 32-bit floats
+            // to a call to `fmodf`, so to work around this we special case
+            // 32-bit MSVC 32-bit float rem instructions and instead do the call
+            // out to `fmod` ourselves.
+            //
+            // Note that this is currently duplicated with src/libcore/ops.rs
+            // which does the same thing, and it would be nice to perhaps unify
+            // these two implementations on day!
+            let use_trait = tcx.sess.target.target.options.is_like_msvc &&
+                            tcx.sess.target.target.arch == "x86" &&
+                            lhs_t == tcx.types.f32;
```

Comment:
This is with optimization turned on?  (The problem is that instcombine narrows frem instructions.)